### ISSUE TITLE
"Maintain focus in the editor" option has no effect when running cells

### ISSUE THREAD

**Issue by pavolgaj** [spyder-ide/spyder#17028]
### What steps reproduce the problem?

1.  Open Tools > Preferences > Editor > Run Code and **enable** the "Maintain focus in the Editor after running cells or selections" option.
2.   Execute Run > Run cell (or Crtl + Return)

### What is the expected output? What do you see instead?

The **focus switches to the console** and doesn't stay in the editor. This is the same behaviour as when the "Maintain focus ..." option is disabled, i.e., enabling the option has no apparent effect. Running line (F9) works fine - it changes focus according to the enabling/disabling this option.
Similar to this issue https://github.com/spyder-ide/spyder/issues/11503. But, now it switches focus always.

## Versions

* Spyder version: 5.2.1
* Python version: Python 3.8.10 64-bit 
* Qt version: Qt 5.12.1
* PyQt version: PyQt5 5.12
* Operating System name/version: Linux 5.4.0-91-generic (Linux Mint)

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  3.0.4 (OK)
cloudpickle >=0.5.0           :  1.5.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.13.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jellyfish >=0.7               :  0.8.9 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  18.0.1 (OK)
nbconvert >=4.0               :  5.6.1 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.6.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.5.1 (OK)
pygments >=2.0                :  2.7.4 (OK)
pylint >=2.5.0                :  2.5.0 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.2;<1.4.0          :  1.3.2 (OK)
pylsp_black >=1.0.0           :  1.0.1 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.0 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  56.0.0 (OK)
sphinx >=0.6.6                :  3.2.0 (OK)
spyder_kernels >=2.2.0;<2.3.0 :  2.2.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  0.10.3 (OK)
xdg >=0.26                    :  0.26 (OK)
zmq >=17                      :  18.1.1 (OK)

# Optional:
cython >=0.21                 :  0.29.21 (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.21.4 (OK)
pandas >=1.1.1                :  1.3.5 (OK)
scipy >=0.17.0                :  1.5.2 (OK)
sympy >=0.7.3                 :  1.9 (OK)
```

**Comment by ccordoba12** [comment]
Hey @pavolgaj, thanks for reporting. Unfortunately, I can't reproduce your error, sorry:

![focus-to-editor](https://user-images.githubusercontent.com/365293/146212692-cca0cd54-994c-4389-84b0-36d347cc1b7e.gif)

As you can see, the focus stays on the editor after running cells in my case, and @dalthviz confirmed he's observing the same behavior on Windows.

**Comment by dalthviz** [comment]
Maybe could you submit here a GIF or video to better understand how this happening to you @pavolgaj ? Let us know!

**Comment by pavolgaj** [comment]
I just tested it on my second laptop (Win7 + Python3.7). Same behaviour. But, when I compare @ccordoba12 video. I see some difference. I have a console and editor in the same part of the main window. And if I run cell, the focus stays in the editor (there is an active cursor there) but the console is on the top and the editor is below it. I didn't notice the active cursor, before. 
If I run a line, the editor is on top.

When I added console and editor separately, the focus stayed in editor like on the video above.

Here is my situation:
![spyder](https://user-images.githubusercontent.com/17177193/146230000-548572c5-1cee-40ab-84e6-4ecff3f52002.gif)
